unison-cli/src/Unison/LSP/Diagnostics.hs

```diff
@@ -38,112 +24,19 @@ uToLspPos uPos =
 uToLspRange :: Range.Range -> Range
 uToLspRange (Range.Range start end) = Range (uToLspPos start) (uToLspPos end)
 
-infoDiagnostics :: FileAnalysis -> Lsp [Diagnostic]
```

Comment:
Rather than analyze the notes from scratch each time for diagnostics, we should analyze the notes once and collect all the info we can from them. So most of this logic just moved to FileAnalysis.hs